raiden/transfer/token_network.py

```diff
@@ -140,11 +197,14 @@ def handle_action_transfer_direct(
         block_number,
 ):
     receiver_address = state_change.receiver_address
-    channel_state = token_network_state.partneraddresses_to_channels.get(receiver_address)
+    channel_states = views.filter_channels_by_status(
+        token_network_state.partneraddresses_to_channels[receiver_address],
+        [CHANNEL_STATE_UNUSABLE],
+    )
 
-    if channel_state:
+    if channel_states:
         iteration = channel.state_transition(
-            channel_state,
+            channel_states[-1],
```

Comment:
if its expected that there will be only one valid state, please add an assert

Reply:
Here we expect to be multiple or at least one. If so, we get the last one. So asserting means that we have to:
```
assert len(channel_states) >= 1
```
Is that what you mean?

Reply:
Here having added a comment why you take the last state with the `-1` would have been good. I know why you do it as we discussed it last week, but I guess our future selves will have questions.